packages/core/saved-objects/core-saved-objects-server-internal/src/apply_type_defaults.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import Semver from 'semver';
+import type { SavedObjectsType } from '@kbn/core-saved-objects-server';
+import { globalSwitchToModelVersionAt } from '@kbn/core-saved-objects-base-server-internal';
+
+/**
+ * Apply global defaults to the provided SO type.
+ */
+export const applyTypeDefaults = (type: SavedObjectsType): SavedObjectsType => {
+  let switchToModelVersionAt = type.switchToModelVersionAt;
+  if (
+    !switchToModelVersionAt ||
+    !Semver.valid(switchToModelVersionAt) ||
+    Semver.gt(switchToModelVersionAt, globalSwitchToModelVersionAt)
```

Comment:
I would normally think of these lines as reasons to throw an error, something like:

`Invalid version detected: "foobar". Please use SemVer compliant versions.` for (1)

`Max switch version is 8.10.0, you provided 8.11.0.` for (2).

In which case we have more of a classic default behaviour. But I get that we may not want to draw attention/discussion quite yet.

Reply:
After thinking about it, I think you're right. It's probably better to throw if the provided version is invalid (bad format) or higher than `8.10.0` (or `8.11.0` if we want to already allow owners to specify a one-version grace period)